File: data/readable_format/cpp/file1327.cpp
// test_ambervar.cpp
#include <gtest/gtest.h>
#include <string>

// Define the ambervar class directly here
class ambervar {
public:
    std::string mmdir;
    std::string qmdir;
    std::string qmbase;
    std::string mmbase;

    // Constructors, destructors, and other member functions can be defined here if necessary.
};

// Unit tests for ambervar
TEST(AmberVarTest, TestSetAndGet) {
    ambervar amberVariables;

    // Set values
    amberVariables.mmdir = "test_mmdir";
    amberVariables.qmdir = "test_qmdir";
    amberVariables.qmbase = "test_qmbase";
    amberVariables.mmbase = "test_mmbase";

    // Test values
    EXPECT_EQ(amberVariables.mmdir, "test_mmdir");
    EXPECT_EQ(amberVariables.qmdir, "test_qmdir");
    EXPECT_EQ(amberVariables.qmbase, "test_qmbase");
    EXPECT_EQ(amberVariables.mmbase, "test_mmbase");
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/readable_format/fortran/file1327.f90
! test_ambervar.f90
module ambervar
    implicit none
    character(len=:), allocatable :: mmdir
    character(len=:), allocatable :: qmdir
    character(len=:), allocatable :: qmbase
    character(len=:), allocatable :: mmbase
end module ambervar

program test_ambervar
    use ambervar
    implicit none

    ! Set values
    mmdir = "test_mmdir"
    qmdir = "test_qmdir"
    qmbase = "test_qmbase"
    mmbase = "test_mmbase"

    ! Test values
    if (mmdir /= "test_mmdir") stop "Test failed for mmdir"
    if (qmdir /= "test_qmdir") stop "Test failed for qmdir"
    if (qmbase /= "test_qmbase") stop "Test failed for qmbase"
    if (mmbase /= "test_mmbase") stop "Test failed for mmbase"

    print *, "All tests passed."
end program test_ambervar